### PR TITLE
[wallet] Keypool topup cleanups

### PR DESCRIPTION
**Pull request by jnewbery**
A couple of minor cleanups suggested by @ryanofsky here: https://github.com/bitcoin/bitcoin/pull/11022#pullrequestreview-55598940

Does not affect functionality. Not required for v0.15.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Tagged for backport if there is an rc2, as this fixes a confusing warning when building from source.

**Comment by paveljanik** [comment]
Warnings removed.

ACK https://github.com/bitcoin/bitcoin/pull/11044/commits/61aa92d9a854677cafefb15117c3d83d6bab6af6

**Comment by jonasschnelli** [comment]
utACK 61aa92d9a854677cafefb15117c3d83d6bab6af6

**Comment by promag** [comment]
Log message is consistent with others related.

Nit, I would reword 063aab4 to `Remove unused CWallet::HasUnusedKeys` since it doesn't really revert c25d90f, but keep the body.

ACK 61aa92d.

**Comment by sipa** [comment]
utACK 61aa92d9a854677cafefb15117c3d83d6bab6af6

**Comment by jnewbery** [comment]
Thanks @promag . I've changed the commit message, but contents of the commit are the same. Check that https://github.com/bitcoin/bitcoin/commit/61aa92d9a854677cafefb15117c3d83d6bab6af6 == 67ceff4039ae038ae16f06128f868a2e8395b59a

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -3659,15 +3659,11 @@ void CWallet::MarkReserveKeysAsUsed(int64_t keypool_id)
             m_pool_key_to_index.erase(keypool.vchPubKey.GetID());
         }
         walletdb.ErasePool(index);
+        LogPrintf("keypool index %d removed\n", index);
```
Comment (laanwj) [inline src/wallet/wallet.cpp]: I'd prefer to put this in the wallet debug category - it's just an info message, and not very useful for most end-users I guess?

Comment (promag) [inline src/wallet/wallet.cpp]: Followup PR to change for all?

**Comment by promag** [comment]
ACK 67ceff4.

File: src/wallet/wallet.cpp (hunk 1 above)
Comment (jnewbery) [inline src/wallet/wallet.cpp]: This matches the existing 'keypool keep' logs, which are not wallet debug category.

This PR already has a bunch of ACKs and it's wanted for rc2, so I'd rather not change this now. I'll happily open a PR to make this (and other wallet logs) wallet debug category.

Comment (laanwj) [inline src/wallet/wallet.cpp]: Ok...
